### ISSUE TITLE
Uncaught exception: OSError: magic_bytes do not match

### ISSUE THREAD

**Issue by kseistrup** [TheKysek/MiNode#2]
Every now and then the following exception is seen:

``` pytb
Exception in thread Connection to ${IP_ADDRESS_AND_PORT}:
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "…/MiNode/src/connection.py", line 76, in run
    self._process_buffer()
  File "…/MiNode/src/connection.py", line 164, in _process_buffer
    h = message.Header.from_bytes(self.buffer[:shared.header_length])
  File "…/MiNode/src/message.py", line 34, in from_bytes
    raise IOError('magic_bytes do not match')
OSError: magic_bytes do not match
```

**Comment by bmng-dev** [comment]
Perhaps this is associated with #4? Abandoning a handshake mid-handshake means MiNode may be inadvertantly processing remnants of the SSL handshake protocol as part of the Bitmessage protocol.

**Comment by TheKysek** [comment]
Resolved by c525d4f2e1ab2ebc62bf5fd7f636c531522af0f7
